src/components/NewComment.jsx

```diff
@@ -0,0 +1,112 @@
+import { useEffect, useState } from "react";
+import { fetchUsers, postComment } from "../utils";
+
+const NewComment = ({ id, setCommentsArr }) => {
+  const [formValues, setFormValues] = useState({
+    username: "",
+    commentBody: "",
+  });
+
+  const [usersArr, setUsersArr] = useState([]);
+  const [feedbackMessage, setFeedbackMessage] = useState({
+    posted: 0,
+    message: "",
+  });
+
+  useEffect(() => {
+    fetchUsers()
+      .then(({ users }) => {
+        return users;
+      })
+      .then((users) => {
+        setUsersArr(users);
+        return users;
+      })
+      .then((users) => {
+        setFormValues((currValue) => {
+          return { ...currValue, username: users[0].username };
+        });
+      });
+  }, []);
+
+  const handleChange = (event) => {
+    if (
+      (event.target.id === "commentBody" && event.target.value.length >= 201) ||
+      event.target.value.startsWith(" ") ||
+      event.target.value.endsWith("  ")
+    ) {
+      return;
+    } else {
+      setFormValues((currValues) => {
+        return { ...currValues, [event.target.id]: event.target.value };
+      });
+    }
+  };
+
+  const submitForm = (event) => {
```

Comment:
We can currently submit a comment twice if we press submit in quick succession.

Also, the new comments appear at the top but appear at the bottom after refreshing.